solver/src/solver/baseline_solver.rs

```diff
@@ -51,6 +56,32 @@ impl BaselineSolvable for ConstantProductOrder {
     }
 }
 
+impl BaselineSolvable for WeightedProductOrder {
```

Comment:
Do we need `BaselineSolvable` for both `WeightedPool` and `WeightedProductOrder`? I see how we do it by converting it and using the other, but it seems a bit redundant.

Reply:
The idea here is that in the `orderbook` price estimates are done using `WeightedPool` directly fetched with a `WeigthedPoolFetching` instance. In the solver, however, we convert everything into a `Liquidity::WeightedProduct` and use that in the baseline solver. For this reason, both need to implement `BaselineSolvable` since both are used by the baseline solving code.

We also do this with `Pool` and `ConstantProductOrder`.